openeo/capabilities.py

```diff
@@ -1,10 +1,92 @@
+import contextlib
 from abc import ABC
-from distutils.version import LooseVersion
+import re
 from typing import Union
 
 # Is this base class (still) useful?
 
 
+class LooseVersion:
+    """A version number consists of a series of numbers,
+    separated by either periods or strings of letters. When comparing
+    version numbers, the numeric components will be compared
+    numerically, and the alphabetic components lexically. The following
+    are all valid version numbers, in no particular order:
+        1.5.1
+        1.5.2b2
+        161
+        3.10a
+        8.02
+        3.4j
+        1996.07.12
+        3.2.pl0
+        3.1.1.6
+        2g6
+        11g
+        0.960923
+        2.2beta29
+        1.13++
+        5.5.kw
+        2.0b1pl0
+    In fact, there is no such thing as an invalid version number under
+    this scheme; the rules for comparison are simple and predictable.
+    """
+
+    component_re = re.compile(r'(\d+ | [a-z]+ | \.)', re.VERBOSE)
+
+    def __init__(self, version_string=None):
+        if isinstance(version_string, str):
+            self.parse(version_string)
+
+        if isinstance(version_string, LooseVersion):
+            self.version = version_string.version
```

Comment:
there is no `else` here: what if `version_string=None` (the default)?

Reply:
fixed